### PR TITLE
Refactor /items tests

### PR DESCRIPTION
**Pull request by RemakingEden**
As discussed we were going to try a new way of testing on the /products (or in this case /items until updated) to see what people thought about it.

I have added:
Fluent assertions - To make the assertions a little more readable
Data driven testing - So we can run the same tests over and over again with different data, e.g. the validation tests
Snapshots - So we don't have to manually write the validation assertions, this saves them into a file that is commited alongside the code
Github Actions - Split the tests into their own workflow for ease as this scales.

I have placed all data we use into a fixtures folder to keep the tests neater.

 Let me know what you think, ive been looking at this for two days now so would be great to get some fresh eyes/opinions.

### CODE REVIEW

File: answerking/settings/base.py [hunk 1]
```diff
@@ -154,3 +154,5 @@
 SECURE_FRAME_DENY = False
 SECURE_CONTENT_TYPE_NOSNIFF = False
 SECURE_BROWSER_XSS_FILTER = False
+
+TEST_RUNNER = 'snapshottest.django.TestRunner'
```
Comment (abbas-khan8) [inline answerking/settings/base.py]: This may need putting in `settings/development.py`

Comment (RemakingEden) [reply answerking/settings/base.py]: This has been moved :)

File: answerking_app/tests/test_items.py [hunk 2]
```diff
@@ -1,247 +1,192 @@
-from django.db.models.query import QuerySet
-from django.test import Client, TestCase, TransactionTestCase
+from django.test import Client, TestCase
+from assertpy import assert_that
+from snapshottest import TestCase
```
Comment (NogaAD) [inline answerking_app/tests/test_items.py]: As you're importing two TestCases maybe renaming it with `as`?

Comment (RemakingEden) [reply answerking_app/tests/test_items.py]: Ive removed the django import now

File: answerking_app/tests/BaseTestClass.py [hunk 3]
```diff
@@ -1,257 +1,183 @@
...
-                    "quantity": quantity,
-                    "subTotal": sub_total,
-                }
-            )
-            order_total += sub_total
 
-        expected_order["lineItems"] = line_items
-        expected_order["orderTotal"] = float(order_total)
-        return expected_order
+class TestBase(TransactionTestCase, TestCase):
```
Comment (PietroConvalleAD) [inline answerking_app/tests/BaseTestClass.py:9]: Should just inherit `TransactionTestCase`

Comment (RemakingEden) [reply answerking_app/tests/BaseTestClass.py:9]: It needs TestCase that is from the snapshotter, not Django. Otherwise snapshotter does not appear to work

**Review by PietroConvalleAD — APPROVED** [review]

**Review by MichaelOSullivanAnswer — APPROVED** [review]

**Pull request merged**